addons/binding/org.openhab.binding.cm11a/src/main/java/org/openhab/binding/cm11a/handler/Cm11aAbstractHandler.java

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.cm11a.handler;
+
+import java.io.IOException;
+
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.PercentType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.cm11a.internal.InvalidAddressException;
+import org.openhab.binding.cm11a.internal.X10Interface;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an abstract base class for the "Thing" handlers (i.e. Cm11aApplianceHandler and Cm11aLampHandler).
+ * It is not used the the Bridge handler (Cm11aHandler)
+ *
+ * @author Bob Raker
+ *
+ */
+public abstract class Cm11aAbstractHandler extends BaseThingHandler {
+
+    /**
+     * The House and Unit codes set on the module, i.e. A1, J14
+     */
+    protected String houseUnitCode;
+
+    /**
+     * The X10 function
+     */
+    protected int x10Function;
+
+    /**
+     * The channel ID
+     */
+    protected ChannelUID channelUID;
+
+    /**
+     * The current State of the device
+     */
+    protected State currentState;
+
+    /**
+     * The "Thing" this device refers to
+     */
+    protected Thing thing;
+
+    /**
+     * Number of CM11a dim increments for dimable devices
+     */
+    static final int X10_DIM_INCREMENTS = 22;
+
+    private Logger logger = LoggerFactory.getLogger(Cm11aAbstractHandler.class);
+
+    /**
+     * The construction
+     *
+     * @param thing The "Thing" to be handled
+     */
+    public Cm11aAbstractHandler(Thing thing) {
+        super(thing);
```

Comment:
Please assign the thing here and make the field private

Reply:
I removed the class variable "thing" because it exists in BaseThingHandler.